packages/hardhat-cannon/src/tasks/deploy.ts

```diff
@@ -102,5 +106,21 @@ task(TASK_DEPLOY, 'Deploy a cannon package to a network')
 
     augmentProvider(hre, outputs);
 
+    if (hre.network.name !== 'hardhat') {
```

Comment:
We also have to check for `'cannon'` network:
```suggestion
    if (hre.network.name !== 'hardhat' && hre.network.name !== 'cannon') {
```